File: apps/hotel-booking-service-admin/src/api/booking/BookingCreateInput.ts
import { CustomerWhereUniqueInput } from "../customer/CustomerWhereUniqueInput";
import { PaymentCreateNestedManyWithoutBookingsInput } from "./PaymentCreateNestedManyWithoutBookingsInput";
import { RoomWhereUniqueInput } from "../room/RoomWhereUniqueInput";

export type BookingCreateInput = {
  checkInDate?: Date | null;
  checkOutDate?: Date | null;
  comment?: string | null;
  customer?: CustomerWhereUniqueInput | null;
  description?: string | null;
  name?: string | null;
  payments?: PaymentCreateNestedManyWithoutBookingsInput;
  room?: RoomWhereUniqueInput | null;
  totalPrice?: number | null;
};

File: apps/dotnet-service/src/APIs/Booking/Dtos/BookingCreateInput.cs
namespace DotnetService.APIs.Dtos;

public class BookingCreateInput
{
    public DateTime? CheckInDate { get; set; }

    public DateTime? CheckOutDate { get; set; }

    public string? Comment { get; set; }

    public DateTime CreatedAt { get; set; }

    public Customer? Customer { get; set; }

    public string? Description { get; set; }

    public string? Id { get; set; }

    public string? Name { get; set; }

    public List<Payment>? Payments { get; set; }

    public Room? Room { get; set; }

    public double? TotalPrice { get; set; }

    public DateTime UpdatedAt { get; set; }
}
